src/core/Akka.Remote.Tests/Transport/AkkaProtocolSpec.cs

```diff
@@ -117,19 +117,18 @@ private Collaborators GetCollaborators()
         private class TestFailureDetector : FailureDetector
         {
 #pragma warning disable CS0420
-            // ReSharper disable once InconsistentNaming
-            internal volatile bool isAvailable = true;
-            public override bool IsAvailable => Volatile.Read(ref isAvailable);
-
-            // ReSharper disable once InconsistentNaming
-            internal volatile bool called;
-            public override bool IsMonitoring => Volatile.Read(ref called);
-#pragma warning restore CS0420
+            private volatile bool _isAvailable = true;
+            public override bool IsAvailable => Volatile.Read(ref _isAvailable);
+            public void SetAvailable(bool available) => Volatile.Write(ref _isAvailable, available);
+            
+            private volatile bool _called;
+            public override bool IsMonitoring => Volatile.Read(ref _called);
 
             public override void HeartBeat()
             {
-                called = true;
+                Volatile.Write(ref _called, true);
```

Comment:
Volatile is being accessed inconsistently, make sure that all volatile are accessed properly.